thoth-app/src/component/reference_modal.rs

```diff
@@ -0,0 +1,697 @@
+        let create_reference = Default::default();
+        let update_reference = Default::default();
+        let notification_bus = NotificationBus::dispatcher();
+        let show_modal_form = ctx.props().show_modal_form;
+        let fetch_reference_fields = Default::default();
+        let reference_fields = Default::default();
+
+        ctx.link().send_message(Msg::GetReferenceFields);
+
+        ReferenceModalComponent {
+            reference,
+            doi,
+            doi_warning,
+            isbn,
+            isbn_warning,
+            in_edit_mode,
+            create_reference,
+            update_reference,
+            notification_bus,
+            show_modal_form,
+            fetch_reference_fields,
+            reference_fields,
+        }
+    }
+
+    fn update(&mut self, ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            Msg::CloseModalForm => {
+                // Prompt parent form to close this form by updating the props
+                // (this will eventually cause this form to re-render)
+                ctx.props().close_modal_form.emit(());
+                false
+            }
+            Msg::ToggleModalFormDisplay => {
+                self.in_edit_mode = ctx.props().reference_under_edit.is_some();
+                if ctx.props().show_modal_form {
+                    if let Some(reference) = ctx.props().reference_under_edit.clone() {
+                        // editing an existing reference
+                        self.reference = reference;
+                    }
+                    // Ensure DOI variable value is kept in sync with reference object.
+                    self.doi = self.reference.doi.clone().unwrap_or_default().to_string();
+                    // Clear DOI warning as the variable value is now valid by definition
+                    // (self.reference.doi can only store valid DOIs)
+                    self.doi_warning = Default::default();
+                    // Ditto for ISBN
+                    self.isbn = self.reference.isbn.clone().unwrap_or_default().to_string();
+                    self.isbn_warning = Default::default();
+                }
+                true
+            }
+            Msg::SetReferenceCreateState(fetch_state) => {
+                self.create_reference.apply(fetch_state);
+                match self.create_reference.as_ref().state() {
+                    FetchState::NotFetching(_) => false,
+                    FetchState::Fetching(_) => false,
+                    FetchState::Fetched(body) => match &body.data.create_reference {
+                        Some(p) => {
+                            // Send newly-created reference to parent form to process
+                            // (parent form is responsible for closing modal)
+                            ctx.props().add_reference.emit(p.clone());
+                            self.reference = Default::default(); // reset form
```

Comment:
I think we should. It was quite annoying, while manually testing, having to reset fields when adding multiple references in one go. I suppose it's bearable in other forms because they don't contain so many fields

Reply:
Sure. Thinking about it some more, other forms not resetting was a combination of a) user convenience (e.g. when adding multiple Related Works in one go, most of the details stay the same) and b) working around #366 (see #229). I think neither of those apply here.